client/src/partials/home/home.js

```diff
@@ -2,58 +2,168 @@ angular.module('bhima.controllers')
 .controller('HomeController', HomeController);
 
 HomeController.$inject = [
-  '$translate', 'CurrencyService', 'ExchangeRateService', 'SessionService', 'SessionService'
+  'CurrencyService', 'ExchangeRateService', 'SessionService', 'SystemService', '$interval', '$translate', '$scope'
 ];
 
-function HomeController($translate, Currencies, Rates, SessionService, Session) {
+/**
+ * Home Controller (system dashboard)
+ *
+ * This controller powers the system dashboard shown by default when the user
+ * signs in. This is currently not very informative since the system is missing
+ * the infrastructure to power the dashboard completely.
+ *
+ * @todo - remove the fake graph generation code and replace with actual system
+ * activity, such as patient registrations.
+ * @todo - implement fiscal year client-side services to get relevant fiscal year
+ * services and information.
+ */
+function HomeController(Currencies, Rates, Session, System, $interval, $translate, $scope) {
   var vm = this;
 
   vm.today = new Date();
-  vm.project = SessionService.project;
-  vm.user = SessionService.user;
-  vm.enterprise = SessionService.enterprise;
-  vm.symbolCurrency = symbolCurrency;
-  // generic error handler
-  function handler(error) {
-    console.log(error);
-  }
 
-  Currencies.read().then(function (data) {
-    vm.currencies = data;
+  // bind the session information
+  vm.project = Session.project;
+  vm.user = Session.user;
+  vm.enterprise = Session.enterprise;
+  vm.graph = {};
+
+  // load exchange rates
+  Currencies.read()
```

Comment:
Could this use the `NotifyService` to tell the user that something has gone wrong? I can't think of why this request would fail but it would not require too much code.